nl/nl_linux.go

```diff
@@ -338,8 +349,9 @@ func (s *NetlinkSocket) Receive() ([]syscall.NetlinkMessage, error) {
 	if nr < syscall.NLMSG_HDRLEN {
 		return nil, fmt.Errorf("Got short response from netlink")
 	}
-	rb = rb[:nr]
-	return syscall.ParseNetlinkMessage(rb)
+	data := make([]byte, nr)
```

Comment:
Doesn't seem like we are saving much if we have to make a new slice right here. Can't you just return syscall.ParseNetlinkMessage(rb[:nr])?  I'm not sure it is worth it to use a pool for this otherwise.